proposals/NNNN-unlock-existential-types-for-all-protocols.md

```diff
@@ -0,0 +1,240 @@
+
+One such case is heterogenous collections, which require value-level abstraction to store their elements of various types:
+
+```swift
+protocol Identifiable {
+  associatedtype ID: Hashable 
+    
+  var id: ID { get }
+}
+
+let naiveIdentifiables: [Identifiable] ❌
+// The compiler doesn't currently allow that.
+// So, we work around that by creating a
+// custom existential type: 'AnyIdentifiable'.
+
+
+struct AnyIdentifiable { 
+  typealias ID = AnyHashable
+    
+  var id: ID { ... }
+}
+
+let myIdentifiables: [AnyIdentifiable] ✅
+```
+
+Furthermore, dynamic environments are also known to lack type information. Therefore value-level abstraction can be exploited in cases such as previewing an application, where the application's components are dynamically replaced, in the file system where a file representing an unknown type might be stored, and in server environments, where various types could be exchanged between different computers.
+
+Morover, the availability of an existential types for a given protocol is sometimes quite unintuitive. That is, today, a protocol qualifies for an existential type provided that it lacks any `Self` or associated type requirements; however, the associated types of a protocol can be fixed via the same-type constraint. As a result, [post](https://forums.swift.org/t/making-a-protocols-associated-type-concrete-via-inheritance/6557) after [post](https://forums.swift.org/t/constrained-associated-type-on-protocol/38770) has been created asking for this restriction's removal:
+
+```swift
+protocol User: Identifiable
+  where ID == UUID {
+  
+  var username: Strng { get }
+}
+  
+let myUsers: [User] ❌
+// This is forbidden today
+// for no apparent reason.
+```
+
+All in all, supporting existential types for all protocols is useful for many situations that involve dynamicity. Not to mention that there are many questions by language users asking about this behavior. Taking everything into consideration, we are confident that addressing this abnormality in the language will build a stronger foundation for [future additions](https://forums.swift.org/t/improving-the-ui-of-generics/22814).
+
+
+### Proposed Solution
+
+The constraint prohibiting the use of some protocols as types will be lifted. Consequently, boilerplate code in many projects - especially libraries and frameworks - will be significantly reduced.
+
+
+### Detailed Design 
+
+The compiler will no longer differentiate between protocols that don’t have `Self` or associated type requirements and those that do. However, some restrictions will apply to the use of requirements referencing associated types as seen in the below examples.
+
+#### Examples:
+
+##### Protocol with `Self` and Associated Type Requirements 
+
+```swift
+protocol Foo {
```

Comment:
Would be nice if all examples used the same style (A, B, etc). The ones below do but this one does not for example